tools/migrate-devspaces-assets.sh

```diff
@@ -0,0 +1,171 @@
+#3. docker       :   https://docs.docker.com/engine/install/
+#4. curl
+#5. gunzip
+#6. tar
+
+HELMDIR=/var/tmp/helm_migrate
+INGRESSNAME=migration-traefik
+ENABLEINGRESS=false
+
+echo ""
+echo "Migrate assets of Azure Dev Spaces to Bridge to Kubernetes"
+echo "----------------------------------------------------------"
+echo ""
+
+helpFunction()
+{
+   echo ""
+   echo "Usage: $1 -g ResourceGroupName -n AKSName -h DockerHubName"
+   echo -e "\t-g Name of resource group of AKS Cluster"
+   echo -e "\t-n Name of AKS Cluster"
+   echo -e "\t-k Kubernetes namespace (uses 'default' namespace otherwise)"
+   echo -e "\t-r Path to Root of the project that needs to be migrated (default = pwd)"
+   echo -e "\t-h Docker Hub repository name"
+   echo -e "\t-t Docker image name & tag in format name:tag (default = projectName:stable)"
+   echo -e "\t-i Enable ingress (default = false)"
+   echo -e "\t-d Helm Debug switch"
+   exit 1 # Exit script after printing help
+}
+
+installHelmFunction()
+{
+  if [ ! -d "$HELMDIR" ]; then
+    echo "Creating directory for helm"
+    mkdir $HELMDIR
+    if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+       curl -fsSL -o $HELMDIR/helm.tar.gz https://get.helm.sh/helm-v3.2.1-linux-amd64.tar.gz
+       gunzip -c $HELMDIR/helm.tar.gz | tar xopf - -C $HELMDIR
+       mv $HELMDIR/linux-amd64/helm $HELMDIR
+    elif [[ "$OSTYPE" == "darwin"* ]]; then
+       curl -fsSL -o $HELMDIR/helm.tar.gz https://get.helm.sh/helm-v3.2.1-darwin-amd64.tar.gz
+       gunzip -c $HELMDIR/helm.tar.gz | tar xopf - -C $HELMDIR
+       mv $HELMDIR/darwin-amd64/helm $HELMDIR
+    else
+       echo "OS not recognized. Please either run on linux-gnu or osx"
+       exit 1
+    fi
+  fi
+}
+
+while getopts "g:n:r:d:k:h:ti" opt; do
+   case "$opt" in
+      g ) RGNAME="$OPTARG"  ;;
+      n ) AKSNAME="$OPTARG"  ;;
+      r ) PROJECTROOT="$OPTARG"  ;;
+      d ) HELMARGS=" --debug" ;;
+      k ) NAMESPACE="$OPTARG" ;;
+      h ) DOCKERREPO="$OPTARG" ;;
+      t ) IMAGENAMEANDTAG="$OPTARG" ;;
+      i ) ENABLEINGRESS="true" ;;
+      ? ) helpFunction ;; # Print helpFunction in case parameter is non-existent
+   esac
+done
+
+# Print helpFunction in case parameters are empty
+if [ -z "$RGNAME" ] || [ -z "$AKSNAME" ] || [ -z "$DOCKERREPO" ]; then
+   echo "Some or all of the parameters are empty";
+   helpFunction
+fi
+
+if [ -z "$NAMESPACE" ]; then
+   NAMESPACE="default"
+fi
+NAMESPACE=$(echo $NAMESPACE | tr [:upper:] [:lower:])
+echo "Using kubernetes namespace: $NAMESPACE"
+
+if [ -z "$PROJECTROOT" ]; then
+   PROJECTROOT="$(pwd)"
+fi
+echo "Using project root: $PROJECTROOT"
+
+echo "Checking directory $PROJECTROOT for 'azds.yaml', 'Dockerfile' & 'charts' folder"
+if [ ! -f "$PROJECTROOT/azds.yaml" ] || [ ! -f "$PROJECTROOT/Dockerfile" ] || [ ! -d "$PROJECTROOT/charts" ]; then
+  echo "$(tput setaf 1)ERROR: '$PROJECTROOT' doesn't seem to be initialized by Azure Dev Spaces. $(tput sgr 0)"
+  echo "Was 'azds prep' ever run in this project?"
```

Comment:
What should we tell the user here? To run azds prep and then run this script? We should be more clear about the next steps here

Reply:
The expectation here is that users would use this only for the projects that are already prep'ed with Azure Dev Spaces.
I can instruct users to run azds prep again and use this script.